x-pack/legacy/plugins/apm/public/projections/base_transaction_groups.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Setup } from '../../server/lib/helpers/setup_request';
+import {
+  PROCESSOR_EVENT,
+  TRANSACTION_NAME
+} from '../../common/elasticsearch_fieldnames';
+import { rangeFilter } from '../../server/lib/helpers/range_filter';
+
+export function getBaseTransactionGroupsProjection({
```

Comment:
I'm not sold yet on the whole `options.type` thing. It feels like using `new Animal({ type: 'dog' })` instead of `new Dog()`. But I can't really think of a situation where that would hurt us today, only when the projections start drifting apart. I'm going to sleep on this one and think about it for a bit 😄

Reply:
I don't have any strong opinions on `options.type` but it feels like the most straightforward. If you can think of another way of combining `base_transaction_groups.ts`, `traces.ts` and `transaction_groups.ts` into a single file that would be great too.

To be clear: I'm merely proposing moving the `options.type` logic from `fetcher.ts` to `getTransactionGroupsProjection` - not adding more of it.